tools/autograd/gen_variable_type.py

```diff
@@ -767,10 +787,41 @@ def declare_returned_variables():
         names = [ret['type'] + ' ' + ret['name'] + ';' for ret in declaration['returns']]
         return '\n'.join(names)
 
-    def wrap_output(call):
-        # Returns `wrapped_call` which is a drop-in replacement for `call`
+    def emit_view_lambda():
+        input_base = 'input_base'
+        replay_view_func = ''
+        updated_unpacked_args = []
+        combined = nested_dict(env, declaration)
+        for arg in combined['unpacked_args']:
+            if arg == 'self_':
+                updated_unpacked_args.append(input_base)
+            elif combined['unpacked_args_simple_type'][arg] == 'IntArrayRef':
+                arg_vec = arg + '_vec'
+                replay_view_func += ARRAYREF_TO_VEC.substitute(arg=arg, vec=arg_vec)
+                updated_unpacked_args.append(arg_vec)
+            else:
+                updated_unpacked_args.append(arg)
+        if 'namespace' in declaration['method_of']:
+            replay_view_call = CALL_DISPATCH_VIA_NAMESPACE.substitute(
+                combined,
+                unpacked_args=updated_unpacked_args)
+        else:
+            replay_view_call = CALL_DISPATCH_VIA_METHOD.substitute(
+                combined,
+                var=input_base,
+                unpacked_method_args=updated_unpacked_args[1:])
```

Comment:
This is copied from

```
            if 'namespace' in declaration['method_of']:
                base_type_call = CALL_DISPATCH_VIA_NAMESPACE.substitute(combined)
            else:
                unpacked_method_args = combined['unpacked_args'][1:]
                base_type_call = CALL_DISPATCH_VIA_METHOD.substitute(
                    combined, unpacked_method_args=unpacked_method_args)
```

Right? Seems worthwhile to factor this into a helper function.